### PR TITLE
Add Strict-Transport-Security header

### PR DESCRIPTION
**Pull request by miklos-martin**
#19861

### CODE REVIEW

File: akka-http-core/src/main/scala/akka/http/impl/model/parser/SimpleHeaders.scala [hunk 1]
```diff
@@ -186,6 +186,10 @@ private[parser] trait SimpleHeaders { this: Parser with CommonRules with CommonA
   // http://tools.ietf.org/html/rfc7231#section-7.4.2
   def server = rule { products ~ EOI ~> (Server(_)) }
 
+  def `strict-transport-security` = rule {
+    ignoreCase("max-age=") ~ `delta-seconds` ~ EOI ~> (`Strict-Transport-Security`(_))
```
Comment (miklos-martin) [inline akka-http-core/src/main/scala/akka/http/impl/model/parser/SimpleHeaders.scala]: I haven't figured out yet how to correctly parse the `includeSubDomains` part too

**Comment by johanandren** [comment]
Hi @miklos-martin, I changed the name of the PR to include [WIP] because as I understand it's not done yet but rather a 'help me out' PR. If I'm wrong, please change back.

**Comment by miklos-martin** [comment]
@johanandren you're right, thanks

**Comment by drewhk** [comment]
OK TO TEST

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala [hunk 2]
```diff
@@ -791,6 +791,19 @@ final case class Server(products: immutable.Seq[ProductVersion]) extends jm.head
   def getProducts: Iterable[jm.headers.ProductVersion] = products.asJava
 }
 
+// https://tools.ietf.org/html/rfc6797
+object `Strict-Transport-Security` extends ModeledCompanion[`Strict-Transport-Security`] {
+  def apply(maxAge: Long, includeSubDomains: Option[Boolean]) = new `Strict-Transport-Security`(maxAge, includeSubDomains.getOrElse(false))
```
Comment (miklos-martin) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala]: I am not quite happy with this, but I couldn't come up with a nicer solution. Can I get some help with this?

**Comment by miklos-martin** [comment]
OK TO TEST

File: akka-docs/rst/scala/http/common/http-model.rst [hunk 3]
```diff
@@ -282,6 +282,16 @@ Connection
...
 
+Strict-Transport-Security
+  HTTP Strict Transport Security (HSTS) is a web security policy mechanism which is communicated by the
+  ``Strict-Transport-Security`` header. The most important security vulnerability that HSTS can fix is SSL-stripping
+  man-in-the-middle attacks. The SSL-stripping attact works by transparently converting a secure HTTPS connection into a
+  plain HTTP connection. The user can see that the connection is insecure, but crucially there is no way of knowing
+  whether the connection should be secure. HSTS addresses this problem by informing the browser that connections to the
+  site should always use TLS/SSL. See also `RFC 6797`_.
+
+.. _RFC 6797: http://tools.ietf.org/html/rfc6797
```
Comment (ktoso) [inline akka-docs/rst/scala/http/common/http-model.rst]: good docs and in the right place, thanks!

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala (hunk 2 above)
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala]: This is fine IMO. Putting the default values inside the parsing would feel even weirder.

**Comment by ktoso** [comment]
Great PR, thanks a lot!

**Comment by ktoso** [comment]
Would you please squash the commits into one? Makes things much simpler to track afterwards, other than that this looks good so let's merge once it's one commit.
Thanks in advance!

**Comment by miklos-martin** [comment]
@ktoso sure, they're squashed

**Comment by ktoso** [comment]
Thanks a lot @miklos-martin !
Will merge after validation

**Pull request merged**